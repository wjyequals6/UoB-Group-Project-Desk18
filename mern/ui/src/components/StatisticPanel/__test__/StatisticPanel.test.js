import React from 'react';
import ReactDOM from 'react-dom';
import StatisticPanel from './../index';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import userEvent from '@testing-library/user-event';


afterEach(cleanup); 

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<StatisticPanel/>, div);
});

it("match snapshot", () => {
    const tree = renderer.create(<StatisticPanel/>).toJSON();
    expect(tree).toMatchSnapshot();
});

