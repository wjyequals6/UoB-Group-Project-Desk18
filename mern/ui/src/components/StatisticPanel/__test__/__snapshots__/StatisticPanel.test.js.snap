// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="site-statistic-demo-card StatisticPanel"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-statistic"
      >
        <div
          className="ant-statistic-title"
        >
          Confirmed
        </div>
        <div
          className="ant-statistic-content"
        >
          <span
            className="ant-statistic-content-value"
          >
            <span
              className="ant-statistic-content-value-int"
            >
              
              29,051
            </span>
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-statistic"
      >
        <div
          className="ant-statistic-title"
        >
          Recovered
        </div>
        <div
          className="ant-statistic-content"
        >
          <span
            className="ant-statistic-content-value"
          >
            <span
              className="ant-statistic-content-value-int"
            >
              
              1,190,188
            </span>
            
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-statistic"
      >
        <div
          className="ant-statistic-title"
        >
          Post
        </div>
        <div
          className="ant-statistic-content"
        >
          <span
            className="ant-statistic-content-value"
          >
            <span
              className="ant-statistic-content-value-int"
            >
              
              0
            </span>
            
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-statistic"
      >
        <div
          className="ant-statistic-title"
        >
          Feedback
        </div>
        <div
          className="ant-statistic-content"
        >
          <span
            className="ant-statistic-content-prefix"
          >
            <span
              aria-label="like"
              className="anticon anticon-like"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="like"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M885.9 533.7c16.8-22.2 26.1-49.4 26.1-77.7 0-44.9-25.1-87.4-65.5-111.1a67.67 67.67 0 00-34.3-9.3H572.4l6-122.9c1.4-29.7-9.1-57.9-29.5-79.4A106.62 106.62 0 00471 99.9c-52 0-98 35-111.8 85.1l-85.9 311H144c-17.7 0-32 14.3-32 32v364c0 17.7 14.3 32 32 32h601.3c9.2 0 18.2-1.8 26.5-5.4 47.6-20.3 78.3-66.8 78.3-118.4 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7 0-12.6-1.8-25-5.4-37 16.8-22.2 26.1-49.4 26.1-77.7-.2-12.6-2-25.1-5.6-37.1zM184 852V568h81v284h-81zm636.4-353l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 16.5-7.2 32.2-19.6 43l-21.9 19 13.9 25.4a56.2 56.2 0 016.9 27.3c0 22.4-13.2 42.6-33.6 51.8H329V564.8l99.5-360.5a44.1 44.1 0 0142.2-32.3c7.6 0 15.1 2.2 21.1 6.7 9.9 7.4 15.2 18.6 14.6 30.5l-9.6 198.4h314.4C829 418.5 840 436.9 840 456c0 16.5-7.2 32.1-19.6 43z"
                />
              </svg>
            </span>
          </span>
          <span
            className="ant-statistic-content-value"
          >
            <span
              className="ant-statistic-content-value-int"
            >
              
              0
            </span>
            
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-card ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-statistic"
          >
            <div
              className="ant-statistic-title"
            >
              Start backing
            </div>
            <div
              className="ant-statistic-content"
              style={
                Object {
                  "color": "#3f8600",
                }
              }
            >
              <span
                className="ant-statistic-content-prefix"
              >
                <span
                  aria-label="arrow-up"
                  className="anticon anticon-arrow-up"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="arrow-up"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M868 545.5L536.1 163a31.96 31.96 0 00-48.3 0L156 545.5a7.97 7.97 0 006 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="ant-statistic-content-value"
              >
                <span
                  className="ant-statistic-content-value-int"
                >
                  
                  9
                </span>
                <span
                  className="ant-statistic-content-value-decimal"
                >
                  .3
                </span>
              </span>
              <span
                className="ant-statistic-content-suffix"
              >
                %
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-col ant-col-12"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <div
        className="ant-card ant-card-bordered"
      >
        <div
          className="ant-card-body"
          style={Object {}}
        >
          <div
            className="ant-statistic"
          >
            <div
              className="ant-statistic-title"
            >
              Stop Jogging
            </div>
            <div
              className="ant-statistic-content"
              style={
                Object {
                  "color": "#cf1322",
                }
              }
            >
              <span
                className="ant-statistic-content-prefix"
              >
                <span
                  aria-label="arrow-down"
                  className="anticon anticon-arrow-down"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="arrow-down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M862 465.3h-81c-4.6 0-9 2-12.1 5.5L550 723.1V160c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v563.1L255.1 470.8c-3-3.5-7.4-5.5-12.1-5.5h-81c-6.8 0-10.5 8.1-6 13.2L487.9 861a31.96 31.96 0 0048.3 0L868 478.5c4.5-5.2.8-13.2-6-13.2z"
                    />
                  </svg>
                </span>
              </span>
              <span
                className="ant-statistic-content-value"
              >
                <span
                  className="ant-statistic-content-value-int"
                >
                  
                  9
                </span>
                <span
                  className="ant-statistic-content-value-decimal"
                >
                  .3
                </span>
              </span>
              <span
                className="ant-statistic-content-suffix"
              >
                %
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
